mig-agent/agent.go

```diff
@@ -783,6 +783,8 @@ func heartbeat(ctx *Context) (err error) {
 		if err != nil {
 			desc := fmt.Sprintf("heartbeat failed with error '%v'", err)
 			ctx.Channels.Log <- mig.Log{Desc: desc}.Err()
+			time.Sleep(ctx.Sleeper)
```

Comment:
Please add a comment above that time.Sleep() referencing this commit or explaining why it's needed. It'll raise eyebrows otherwise.